{
    "issue": {
        "title": "pa.array type inference bug when using NS-timestamp",
        "body": "***Note**: This issue was originally created as [ARROW-2554](https://issues.apache.org/jira/browse/ARROW-2554). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*\n\n### Original Issue Description:\nThe following fails:\r\n```\n\r\nimport pandas as pd\r\nimport pyarrow as pa\r\n\r\npa.array([pd.Timestamp('now').to_datetime64()])\r\n```\r\n\r\nwith `ArrowNotImplementedError: Cannot convert NumPy datetime64 objects with differing unit`, but when you provide the correct type information directly, it works:\r\n\r\n```\n\r\nimport pandas as pd\r\nimport pyarrow as pa\r\n\r\npa.array([pd.Timestamp('now').to_datetime64()], type=pa.timestamp('ns'))\r\n```",
        "created_at": "2018-05-08T15:51:00.000Z",
        "updated_at": "2018-06-14T12:21:26.000Z",
        "labels": [
            "Migrated from Jira",
            "Component: Python",
            "Type: bug"
        ],
        "closed": true,
        "closed_at": "2018-06-14T12:21:24.000Z"
    },
    "comments": [
        {
            "created_at": "2018-06-14T12:21:24.777Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2554?focusedCommentId=16512391) by Uwe Korn (uwe):*\nIssue resolved by pull request 2118\n<https://github.com/apache/arrow/pull/2118>"
        }
    ]
}